File: erlang/fizzbuzz/src/fizzbuzz.erl
-module(fizzbuzz).

-compile(export_all).

fizzbuzz_of(N) when (N rem 15 == 0) -> 
  "fizzbuzz";
fizzbuzz_of(N) when (N rem 3 == 0) -> 
  "fizz";
fizzbuzz_of(N) when (N rem 5 == 0) -> 
  "buzz";
fizzbuzz_of(N) -> 
  N.

to(N) -> 
  lists:foreach(fun(M) -> io:format("~p ", [fizzbuzz_of(M)]) end , lists:seq(1,N)).

main(Args) ->
  to(list_to_integer(lists:nth(1, Args))).

File: scala/fizzbuzz/src/main/scala/FizzBuzz.scala
object FizzBuzz {
  def main(args: Array[String]): Unit = {
    println("Fizz buzz from 1 to 100")
    (1 to 100).foreach(num => println(s"$num. - ${of(num)}"))
  }

  def of(value: Int): String =
    (value % 5, value % 3) match {
      case (0, 0) => "fizzbuzz"
      case (_, 0) => "fizz"
      case (0, _) => "buzz"
      case _ => value.toString
    }
}
